lib/usd/utils/util.cpp

```diff
@@ -102,6 +104,46 @@ void replaceReferenceItems(
         }
     }
 }
+
+// This template method updates the SdfPath for inherited or specialized arcs 
+// when the path to the concrete prim they refer to has changed.
+// HS January 13, 2021: Find a better generic way to consolidate this method with replaceReferenceItems
+template<typename T>
+void replacePath(
+    const UsdPrim&            oldPrim,
+    const SdfPath&            newPath,
+    const T&                  proxy,
+    SdfListOpType             op)
+{
+    // set the listProxy based on the SdfListOpType
+    T::ListProxy listProxy = proxy.GetAppendedItems();
+    if (op == SdfListOpTypePrepended) {
+        listProxy = proxy.GetPrependedItems();
+    } else if (op == SdfListOpTypeOrdered) {
+        listProxy = proxy.GetOrderedItems();
+    } else if (op == SdfListOpTypeAdded) {
+        listProxy = proxy.GetAddedItems();
+    } else if (op == SdfListOpTypeDeleted) {
+        listProxy = proxy.GetDeletedItems();
+    }
+
+    for (const SdfPath path : listProxy) {
+        SdfPath finalPath;
+        if (oldPrim.GetPath() == path.GetPrimPath()) {
+            finalPath = newPath;
+        } else if (path.GetPrimPath().HasPrefix(oldPrim.GetPath())) {
+            finalPath = path.GetPrimPath().ReplacePrefix(oldPrim.GetPath(), newPath);
+        }
+
+        if (finalPath.IsEmpty()) {
+            continue;
+        }
+
+        // replace the old SdfPath with new one
+        listProxy.Replace(path, finalPath);
+    }
+}
```

Comment:
Unlike SdfReferencesProxy that deals with SdfReference, SdfInheritsProxy/SdfSpecializesProxy deal with SdfPath.

I would have liked to consolidate both "replacePath" and "replaceInternalReferencePath" methods since there is a little bit of code duplication. Haven't found a clean way to do it.